packages/core/src/api/message.ts

```diff
@@ -0,0 +1,182 @@
+import { makeURLSearchParams, type RawFile, type REST } from '@discordjs/rest';
+import {
+	Routes,
+	type APIMessage,
+	type APIUser,
+	type RESTGetAPIChannelMessageReactionUsersQuery,
+	type RESTPostAPIChannelMessageJSONBody,
+} from 'discord-api-types/v10';
+
+export class MessagesAPI {
+	public constructor(private readonly rest: REST) {}
+
+	/**
+	 * Sends a message in a channel
+	 *
+	 * @param channelId - The id of the channel to send the message in
+	 * @param message - The options to use when sending the message
+	 */
+	public async send(channelId: string, message: RESTPostAPIChannelMessageJSONBody & { files?: RawFile[] }) {
+		const messageOptions = typeof message === 'string' ? { content: message } : message;
+
+		return (await this.rest.post(Routes.channelMessages(channelId), {
+			files: messageOptions.files,
+			body: messageOptions,
```

Comment:
The files are being sent in the body as well. Should probably change that (can destructure them in the parameters)